irohad/torii/impl/command_service_impl.cpp

```diff
@@ -174,62 +174,38 @@ namespace iroha {
       const auto status_issuer = "ToriiBatchProcessor";
       const auto &txs = batch->transactions();
 
-      bool cache_has_at_least_one_tx{false};
-      bool batch_has_mst_pending_tx{false};
-      std::tie(cache_has_at_least_one_tx, batch_has_mst_pending_tx) =
-          // this accumulate can be split on two parts to perform visit_in_place
-          // two times - one for cache lookup with booleans initialization and
-          // another for statuses pushing. That can allow to move a part of code
-          // to a separate method for simplification
-          std::accumulate(
-              txs.begin(),
-              txs.end(),
-              std::make_pair<bool, bool>(false, false),
-              [this, &status_issuer](std::pair<bool, bool> lookup_result,
-                                     const auto &tx) {
-                const auto &tx_hash = tx->hash();
-                if (auto found = cache_->findItem(tx_hash)) {
-                  iroha::visit_in_place(
-                      (*found)->get(),
-                      [this, &found, &lookup_result, &status_issuer](
-                          const shared_model::interface::MstPendingResponse &) {
-                        this->pushStatus(status_issuer, *found);
-                        lookup_result.second = true;
-                      },
-                      [this, &tx_hash, &status_issuer](
-                          const shared_model::interface::NotReceivedTxResponse
-                              &) {
-                        // This branch covers an impossible case (this cache
-                        // cannot contain NotReceivedTxResponse, because the tx
-                        // has reached processBatch method, which means that
-                        // the tx already has StatelessValid status).
-                        // That is why we are not updating its status inside
-                        // internal cache, but still pushing to status bus.
-                        this->pushStatus(
-                            status_issuer,
-                            status_factory_->makeStatelessValid(tx_hash));
-                      },
-                      [this, &found, &status_issuer](const auto &status) {
-                        this->pushStatus(status_issuer, *found);
-                      });
-                  lookup_result.first = true;
-                }
-                return lookup_result;
-              });
+      bool has_final_status{false};
+      has_final_status = std::accumulate(
```

Comment:
Could you rewrite it with `for` or `std::any_of`? `std::accumulate` with early-return lambda looks somewhat ill-formed.